internal/subscriber/core_test.go

```diff
@@ -0,0 +1,177 @@
+	"github.com/razorpay/metro/internal/brokerstore/mocks"
+	"github.com/razorpay/metro/internal/credentials"
+	"github.com/razorpay/metro/internal/offset"
+	oCore "github.com/razorpay/metro/internal/offset/mocks/core"
+	"github.com/razorpay/metro/internal/subscription"
+	sCore "github.com/razorpay/metro/internal/subscription/mocks/core"
+	"github.com/razorpay/metro/pkg/cache"
+	mCache "github.com/razorpay/metro/pkg/cache/mocks"
+	"github.com/razorpay/metro/pkg/messagebroker"
+	mockMB "github.com/razorpay/metro/pkg/messagebroker/mocks"
+	"github.com/stretchr/testify/assert"
+
+	"testing"
+)
+
+func TestCore_NewSubscriber(t *testing.T) {
+	ctrl := gomock.NewController(t)
+	mockBrokerStoreCore := mocks.NewMockIBrokerStore(ctrl)
+	mockMessageBroker := mockMB.NewMockBroker(ctrl)
+	mockSubscriptionCore := sCore.NewMockICore(ctrl)
+	mockOffsetCore := oCore.NewMockICore(ctrl)
+	ch := mCache.NewMockICache(ctrl)
+	ackCh := make(chan *AckMessage)
+	modAckCh := make(chan *ModAckMessage)
+	requestCh := make(chan *PullRequest)
+	ctx := context.Background()
+	sub := subscription.Model{
+		Name:  subName,
+		Topic: topicName,
+		PushConfig: &subscription.PushConfig{
+			PushEndpoint: "https://www.razorpay.com/api",
+			Attributes: map[string]string{
+				"test_key": "test_value",
+			},
+			Credentials: &credentials.Model{
+				Username: "test_user",
+				Password: "",
+			},
+		},
+		DeadLetterPolicy:               nil,
+		RetryPolicy:                    nil,
+		AckDeadlineSeconds:             10,
+		ExtractedTopicProjectID:        "project123",
+		ExtractedTopicName:             topicName,
+		ExtractedSubscriptionName:      subName,
+		ExtractedSubscriptionProjectID: "project123",
+	}
+	cs := getMockConsumerCore(ctx, ctrl)
+	type fields struct {
+		bs               brokerstore.IBrokerStore
+		subscriptionCore subscription.ICore
+		offsetCore       offset.ICore
+		ch               cache.ICache
+	}
+	type args struct {
+		ctx                    context.Context
+		subscriberID           string
+		subscription           *subscription.Model
+		timeoutInMs            int
+		maxOutstandingMessages int64
+		maxOutstandingBytes    int64
+		requestCh              chan *PullRequest
+		ackCh                  chan *AckMessage
+		modAckCh               chan *ModAckMessage
+	}
+	tests := []struct {
+		name    string
+		fields  fields
+		args    args
+		wantErr bool
+	}{
+		{
+			name: "Create new subscriber basicImpl",
+			fields: fields{
+				bs:               mockBrokerStoreCore,
+				subscriptionCore: mockSubscriptionCore,
+				offsetCore:       mockOffsetCore,
+				ch:               ch,
+			},
+			args: args{
+				ctx:                    ctx,
+				subscriberID:           subID,
+				subscription:           &sub,
+				timeoutInMs:            1000,
+				maxOutstandingMessages: 1000,
+				maxOutstandingBytes:    1000,
+				requestCh:              requestCh,
+				ackCh:                  ackCh,
+				modAckCh:               modAckCh,
+			},
+			wantErr: false,
+		},
```

Comment:
@sumitkrzp can we add another test case where `wantErr: true`

Reply:
Added test case for wantErr true